src/altinn-app-frontend/src/features/form/dynamics/types/index.ts

```diff
@@ -12,6 +12,7 @@ export interface IConditionalRenderingRule {
 
 export interface IConditionalRenderingRepeatingGroup {
   groupId: string;
+  childGroupId?: string;
```

Comment:
Does the app developer have to specify this in the configuration?

Reply:
Yeah, I just extended the existing setup.

But it is a point that we could potentially just fetch the group id's by manipulating the `inputParams` and looking for the group with the matching `group`-databinding in the layout files. 

Given the example input-from the test:

```
        inputParams: {
          number: 'parentGroup{0}.childGroup{1}.mockField',
        },
```

We could do some manipulation here and extract `parentGroup` and `parentGroup.childGroup` strings and find the matching group-component from the layout.

This would maybe be smoother for the app-developer in one sense. But they still would have to define these index indicators manually - so I'm not really sure how much value it would add. The group-id's are probably the easiest thing about this configuration. Could also be made smoother if we have a better GUI in studio to make this configuration a little nicer.

Any thoughts? Worth looking into?

Reply:
I guess my worry is that the config for dynamics is already quite complicated. But, this is in part because the GUI in studio isn't great, and a lot has to be done manually.
I think that improving the GUI in studio is something that we want to do anyway. Auto-populating the group/child group fields is something we might do to make app developer experience simpler, so I think we can leave this as is for now.